src/node_v8.cc

```diff
@@ -295,6 +295,154 @@ static void SetHeapStatistics(JSONWriter* writer, Isolate* isolate) {
+          FIXED_ONE_BYTE_STRING(isolate, "usedSizeBytes"),
+          FIXED_ONE_BYTE_STRING(isolate, "objectStatistics")};
+      Local<v8::Value> pageStatsValues[] = {
+          ToV8Value(context, pageStats.committed_size_bytes, isolate)
+              .ToLocalChecked(),
+          ToV8Value(context, pageStats.resident_size_bytes, isolate)
+              .ToLocalChecked(),
+          ToV8Value(context, pageStats.used_size_bytes, isolate)
+              .ToLocalChecked(),
+          Array::New(isolate, objectStatisticsArray.data(), objectStatisticsArray.size())};
+      Local<Object> pageStatsObject = Object::New(
+          isolate, v8::Null(isolate), pageStatsNames, pageStatsValues, 4);
+
+      pageStatisticsArray.emplace_back(pageStatsObject);
+    }
+
+    // Free List Statistics
+    Local<v8::Name> freeListStatisticsNames[] = {
+        FIXED_ONE_BYTE_STRING(isolate, "bucketSize"),
+        FIXED_ONE_BYTE_STRING(isolate, "freeCount"),
+        FIXED_ONE_BYTE_STRING(isolate, "freeSize")};
+    Local<v8::Value> freeListStatisticsValues[] = {
+        ToV8Value(context, spaceStats.free_list_stats.bucket_size, isolate)
+            .ToLocalChecked(),
+        ToV8Value(context, spaceStats.free_list_stats.free_count, isolate)
+            .ToLocalChecked(),
+        ToV8Value(context, spaceStats.free_list_stats.free_size, isolate)
+            .ToLocalChecked()};
+    Local<Object> freeListStatsObj = Object::New(isolate,
+                                                 v8::Null(isolate),
+                                                 freeListStatisticsNames,
+                                                 freeListStatisticsValues,
+                                                 3);
+
+    // Set Space Statistics
+    Local<v8::Name> spaceStatsNames[] = {
+        FIXED_ONE_BYTE_STRING(isolate, "name"),
+        FIXED_ONE_BYTE_STRING(isolate, "committedSizeBytes"),
+        FIXED_ONE_BYTE_STRING(isolate, "residentSizeBytes"),
+        FIXED_ONE_BYTE_STRING(isolate, "usedSizeBytes"),
+        FIXED_ONE_BYTE_STRING(isolate, "pageStats"),
+        FIXED_ONE_BYTE_STRING(isolate, "freeListStats")};
+    Local<v8::Value> spaceStatsValues[] = {
+        ToV8Value(context, stats.space_stats[i].name.c_str(), isolate)
+            .ToLocalChecked(),
+        ToV8Value(context, stats.space_stats[i].committed_size_bytes, isolate)
+            .ToLocalChecked(),
+        ToV8Value(context, stats.space_stats[i].resident_size_bytes, isolate)
+            .ToLocalChecked(),
+        ToV8Value(context, stats.space_stats[i].used_size_bytes, isolate)
+            .ToLocalChecked(),
+        Array::New(
+            isolate, pageStatisticsArray.data(), pageStatisticsArray.size()),
+        freeListStatsObj,
+    };
+    Local<Object> spaceStatsObject = Object::New(
+        isolate, v8::Null(isolate), spaceStatsNames, spaceStatsValues, 6);
+    spaceStatisticsArray.emplace_back(spaceStatsObject);
+  }
+
+  std::string detailLevel =
```

Comment:
Since this comes from JS land originally, it's probably better to just assign this field in JS land and skip this field in C++, which saves some string transcoding cost.